doc/netplan.md

```diff
@@ -630,6 +630,14 @@ wpasupplicant installed if you let the ``networkd`` renderer handle wifi.
           and ``adhoc`` (peer to peer networks without a central access point).
           ``ap`` is only supported with NetworkManager.
 
+``wowlan`` (sequence of scalars)
+
+:    This enables WakeOnWLan on supported devices. Not all drivers support all
+     options. May be any combination of ``any``, ``disconnect``, ``magic_pkt``,
+     ``gtk_rekey_failure``, ``eap_identity_req``, ``four_way_handshake``,
+     ``rfkill_release``, ``tcp`` or ``ignore``. The value of ``tcp`` and
+     ``ignore`` is only supported with NetworkManager.
```

Comment:
These are the names used in the kernel and in wpasupplicant and AIUI there are no other implementors, so I think it is best to be consistent with the existing "well known" names rather than trying to reinvent.

What surprises me is that there are NM-specific options.  How is 'tcp' implemented in NM, given that it's not in the kernel list - does the kernel have to wake up all the way, trigger NM, which then shuts the system back down?  I would suggest that if so, we should maybe not expose such an NM-specific option which seems hackish.

Also, what does 'ignore' mean when included in a list of all of these others?  Isn't "ignore" simply the absence of any of the other options, and should be considered an empty default?

Reply:
`tcp` is an actual Linux kernel/nl80211 flag (NL80211_WOWLAN_TRIG_TCP_CONNECTION), which is just not implemented by wpa_supplicant. Thus, I kept this flag (as NM-only).

I removed the `ignore` flag, as this defines a fallback to the kernel defaults, while 'default' defines a fallback to the NM-global defaults (from `/etc/NetworkManager/NetworkManager.conf`), which are usually empty (e.g. fallback to kernel defaults, too).